docs/reference/configuration/index.md

```diff
@@ -0,0 +1,236 @@
+---
+id: index
+title: "Introduction to Configuration in ZIO"
+---
+
+Configuration is a core concern for any cloud-native application. So ZIO ships with built-in support for configuration by providing a front-end for configuration providers as well as metrics and logging.
+
+So, ZIO provides a unified way to configure our applications, while still enabling customizability, flexibility, and significant integrations with configuration backends via ecosystem projects, most notably [ZIO Config](https://zio.dev/zio-config).
+
+This configuration front-end allows ecosystem libraries and applications to declaratively describe their configuration needs and delegates the heavy lifting to a ConfigProvider, which may be supplied by third-party libraries such as ZIO Config.
+
+The ZIO Core ships with a simple default config provider, which reads configuration data from environment variables and if not found, from system properties. This can be used for development purposes or to bootstrap applications toward more sophisticated config providers.
+
+## Primitive Configs
+
+ZIO provides a set of primitive configs for the most common types like `int`, `long`, `string`, `boolean`, `double`, etc. All of these configs are available inside the `Config` object.
+
+Let's start with a simple example of how to read configuration from environment variables and system properties:
+
+```scala mdoc:compile-only
+import zio._
+
+object MainApp extends ZIOAppDefault {
+  def run = {
+    for {
+      host <- ZIO.config(Config.string("host"))
```

Comment:
No where is it mentioned that `ZIO.config` allows you to load config from the current config provider.

We should explain things in terms of:

 - `Config[A]` allowing you to describe configuration data modeled by some Scala type `A`
 - `ZIO.config` allowing you to load configuration data described by `Config`
 - `ConfigProvider` being the "engine" that `ZIO.config` uses to load config
 - `Runtime.set/addConfigProvider` being the high-level ways to set or add config providers using layers

Then separately discuss best practices

 - Best practice to put `Config` value into companion object of data type and call it `config`.
 - Best practice for services to load their own config using `ZIO.config` in the layer constructor (NOT for services to require configuraiton as layer input!!! Old deprecated pattern)

Reply:
I added getting started section to cover these topics.